phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4168,4 +4176,197 @@ public MutationState useSchema(UseSchemaStatement useSchemaStatement) throws SQL
+            } else {
+                revokePermissionsFromUser(hConnection, revokeStatement);
+            }
+
+        } catch (SQLException e) {
+            // Bubble up the SQL Exception
+            throw e;
+        } catch (Throwable throwable) {
+            // Wrap around other exceptions to PhoenixIOException (Ex: org.apache.hadoop.hbase.security.AccessDeniedException)
+            throw ServerUtil.parseServerException(throwable);
+        }
+
+        return new MutationState(0, 0, connection);
+    }
+
+
+    private void revokePermissionsFromTables(HConnection hConnection, RevokeStatement revokeStatement, PTable inputTable) throws Throwable {
+
+        org.apache.hadoop.hbase.TableName tableName = SchemaUtil.getPhysicalTableName
+                (inputTable.getName().getBytes(), inputTable.isNamespaceMapped());
+
+        revokePermissionsFromTable(hConnection, revokeStatement, tableName);
+
+        for(PTable indexTable : inputTable.getIndexes()) {
+            // Local Indexes don't correspond to new physical table, they are just stored in separate CF of base table.
+            if(indexTable.getIndexType().equals(IndexType.LOCAL)) {
+                continue;
+            }
+            logger.info("Revoking perms from IndexTable: " + indexTable.getName() + " BaseTable: " + inputTable.getName());
+            if (inputTable.isNamespaceMapped() != indexTable.isNamespaceMapped()) {
+                throw new TablesNotInSyncException(inputTable.getTableName().getString(),
+                        indexTable.getTableName().getString(), "Namespace properties");
+            }
+            tableName = SchemaUtil.getPhysicalTableName(indexTable.getName().getBytes(), indexTable.isNamespaceMapped());
+            revokePermissionsFromTable(hConnection, revokeStatement, tableName);
+        }
+
+        byte[] viewIndexTableBytes = MetaDataUtil.getViewIndexPhysicalName(inputTable.getPhysicalName().getBytes());
+        tableName = org.apache.hadoop.hbase.TableName.valueOf(viewIndexTableBytes);
+        boolean viewIndexTableExists = connection.getQueryServices().getAdmin().tableExists(tableName);
+        if(!viewIndexTableExists && inputTable.isMultiTenant()) {
+            logger.error("View Index Table not found for MultiTenant Table: " + inputTable.getName());
+            throw new TablesNotInSyncException(inputTable.getTableName().getString(),
+                    Bytes.toString(viewIndexTableBytes), " View Index table should exist for MultiTenant tables");
+        }
+        if(viewIndexTableExists) {
+            revokePermissionsFromTable(hConnection, revokeStatement, tableName);
+        }
+
+    }
+
+    private void revokePermissionsFromTable(HConnection hConnection, RevokeStatement revokeStatement, org.apache.hadoop.hbase.TableName tableName)
+            throws Throwable {
+        AccessControlClient.revoke(hConnection, tableName, revokeStatement.getName(),
+                null, null, Permission.Action.values());
+    }
```

Comment:
Avoid using deprecated classes like HConnection.

Reply:
I will look into the alternate. Will refactor.